cyberdrop_dl/config_definitions/config_settings.py

```diff
@@ -10,7 +10,7 @@
 from cyberdrop_dl.utils.data_enums_classes.hash import Hashing
 from cyberdrop_dl.utils.data_enums_classes.supported_domains import SUPPORTED_SITES_DOMAINS
 
-from .pydantic.custom_types import AliasModel, HttpAppriseURL, NonEmptyStr
+from .pydantic.custom_types import AliasModel, HttpAppriseURL, HttpURL, NonEmptyStr
```

Comment:
This needs to be the standard `HttpUrl` cause the check are done as normal strings
```suggestion
from pydantic import HttpUrl
from .pydantic.custom_types import AliasModel, HttpAppriseURL, NonEmptyStr
```

Reply:
I checked the pydantic docs but `HttpUrl` does not actually inherit from `str`. Needs to also add a field validator with mode after to convert them all to strings or just manually convert every URL before checking

Reply:
Relevant: https://github.com/pydantic/pydantic/discussions/6395#discussioncomment-7159870

Reply:
Actually, I think this one is better: https://github.com/pydantic/pydantic/discussions/6395#discussioncomment-7361416